docs/src/main/asciidoc/security-jpa.adoc

```diff
@@ -0,0 +1,360 @@
+====
+
+This command generates a Maven project, importing the `elytron-security-jpa` extension
+which allows you to map your security source to JPA entities.
+
+== Writing the application
+
+Let's start by implementing the `/api/public` endpoint. As you can see from the source code below, it is just a regular JAX-RS resource:
+
+[source,java]
+----
+package org.acme.elytron.security.jpa;
+
+import javax.annotation.security.PermitAll;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/api/public")
+public class PublicResource {
+
+    @GET
+    @PermitAll
+    @Produces(MediaType.TEXT_PLAIN)
+    public String publicResource() {
+        return "public";
+   }
+}
+----
+
+The source code for the `/api/admin` endpoint is also very simple. The main difference here is that we are using a `@RolesAllowed` annotation to make sure that only users granted with the `admin` role can access the endpoint:
+
+
+[source,java]
+----
+package org.acme.elytron.security.jpa;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/api/admin")
+public class AdminResource {
+
+    @GET
+    @RolesAllowed("admin")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String adminResource() {
+         return "admin";
+    }
+}
+----
+
+Finally, let's consider the `/api/users/me` endpoint. As you can see from the source code below, we are trusting only users with the `user` role.
+We are using `SecurityContext` to get access to the current authenticated Principal and we return the user's name. This information is loaded from the database.
+
+[source,java]
+----
+package org.acme.elytron.security.jpa;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.SecurityContext;
+
+@Path("/api/users")
+public class UserResource {
+
+    @GET
+    @RolesAllowed("user")
+    @Path("/me")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String me(@Context SecurityContext securityContext) {
+        return securityContext.getUserPrincipal().getName();
+    }
+}
+----
+
+=== Defining our user entity
+
+We can now describe how our security information is stored in our model by adding a few annotations to our `User` entity:
+
+[source,java]
+----
+package org.acme.elytron.security.jpa;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntity;
+import io.quarkus.elytron.security.common.runtime.BcryptUtil;
+import io.quarkus.security.jpa.runtime.Password;
+import io.quarkus.security.jpa.runtime.Roles;
+import io.quarkus.security.jpa.runtime.UserDefinition;
+import io.quarkus.security.jpa.runtime.Username;
```

Comment:
Everything that is user API should be at the root, not in the `runtime` package. `runtime` is an internal package.

The package should also be `elytron.security.jpa` to be consistent with the other Elytron extensions.

Reply:
Oh, I didn't know that. Will move.